app/models/user.rb

```diff
@@ -87,4 +88,17 @@ def send_devise_notification(notification, *args)
   def decorate
     UserDecorator.new(self)
   end
+
+  private
+
+  # used by zxcvbn
+  def weak_words
+    [APP_NAME]
+  end
+
+  # method required by zxcvbn
+  def password_required?
+    confirmable = confirmed? || confirmation_token.present?
+    password.present? && confirmable
+  end
```

Comment:
Ugh. I wish this pattern of polluting the model would stop. This gem forces us to ask `password_required?` every time the User is validated, even when we know it isn't required.

Could we hardcode this to `false` here, and move the gem's validation to our form object?